### ISSUE TITLE
Goimports "local" mode

### ISSUE THREAD

**Issue by AlekSi** [golangci/golangci-lint#209]
Specifically, `-local` flag:
>  -local string
>    	put imports beginning with this string after 3rd-party packages

That is very useful for projects that follow this grouping style:
```go
import (
  // stdlib

  // third-party

  // other packages of that project
)
```

**Comment by AlekSi** [comment]
Yes, but I want it to complain if imports are not formatted with `-local` flag, but `local` setting is present in the configuration.

**Comment by jirfag** [comment]
Hi!
Thank you for the idea,
I supported it in the master branch. I will include it into a next release

**Comment by thepwagner** [comment]
@jirfag I don't think this is working as intended yet.

We're setting https://github.com/golangci/golangci-lint/blob/master/pkg/golinters/gofmt.go#L115 , which resolves to `golang.org/x/tools/imports.LocalPrefix` ( https://github.com/golangci/golangci-lint/blob/master/pkg/golinters/gofmt.go#L9 )

Then we invoke https://github.com/golangci/golangci-lint/blob/master/vendor/github.com/golangci/gofmt/goimports/golangci.go#L11 , which in turn uses https://github.com/golangci/golangci-lint/blob/master/vendor/github.com/golangci/gofmt/goimports/golangci.go#L8.

So when GoImports is invoked, it doesn't have the `LocalPrefix` that we attempted to set.

I won't PR this as it's pretty deep into dep hell - swapping https://github.com/golangci/golangci-lint/blob/master/pkg/golinters/gofmt.go#L9 to `github.com/golangci/tools/imports` works for me - but there may be a better solution (or more cleanup: like picking one of the vendored `goimports` and using it).

**Comment by jirfag** [comment]
thank you
